src/pipe.js

```diff
@@ -46,21 +46,46 @@ var Pipe = (function (doc, perf) { //eslint-disable-line no-unused-vars, strict
             start.parentNode.removeChild(start);
             end.parentNode.removeChild(end);
             script && require([script], function (i) {
+                // Exposed fragment initialization Function/Promise
                 var init = i && i.__esModule ? i.default : i;
-                if (typeof init === 'function') {
-                    // capture initializaion cost of each fragment on the page
-                    if (perf && 'mark' in perf) {
-                        perf.mark(fragmentId);
-                        init(node);
+                // early return
+                if (typeof init !== 'function') {
+                    return;
+                }
+                // check if User Timing API is supported
+                var isUTSupported = perf && 'mark' in perf;
+
+                function isPromise(obj) {
```

Comment:
maybe, but it's probably something that should be covered by a test, so pulling it out into a library func is going to help with that.